### PR TITLE
nrf5: Enlarge stack to fix thread join overruns

### LINKED ISSUE #2187: nrf5 examples stop responding while attempting thread join
As of https://github.com/openthread/openthread/pull/5299 thread joiner initiated at boot results in unresponsive device. It takes several join attempts for this to occur.

Repro (fails):
```
  git checkout b15c292ee56a0505d3a3904eb295b38bac71cfc9
  git submodule update --init

  cd third_party/openthread/repo
  git checkout addb1936c3e7e14e59f617751e595596980e3ec8 # fails
  cd examples/lock-app/nrf5
  make flash
  
  # thread CLI
  factoryreset
```

  Boot and just wait. Device hangs after a few join attempts.

No Repro: (stable)
```
  cd third_party/openthread/repo
  git revert HEAD

  cd examples/lock-app/nrf5
  make flash
```
  Boot and just wait. Device stable.

Log:
```
[00000000] <info> chip: [DL] Setup PIN discriminator not found; using default: f00
[00000000] <info> chip: [DL] Joiner Discerner: 3840
[00000000] <info> chip: [DL] Setup PIN code not found; using default: 12345678
[00000000] <info> chip: [DL] Joiner PSKd: 12345678
[00000000] <info> chip: [DL] Joiner start: No Error
[00000000] <debug> chip: [DL] Thread joiner timer triggered: No Error
<hung>
```

@lanyuwen @bukepo

### PR DESCRIPTION
**Pull request by mspang**
As of b15c292ee ("[nrf5-lock] start joiner role on boot (#1962)"),
we are using too much stack space in timer task. The timer task has a 1k
stack and logging alone uses a 256 byte stack buffer.

The code in
`GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::OnJoinerTimer `should
be moved off the timer task. In the meantime increase the stack size
to avoid overruns in the thread joiner.

Also enable the option `configCHECK_FOR_STACK_OVERFLOW`, and while we're
here also enable `configUSE_MALLOC_FAILED_HOOK`. These diagnostic options
are invaluable for saving debugging time.

Since logging uses significant stack space, try to catch stack overflows
in the platform `LogV()`. This fires pretty reliably in `OnJoinerTimer` prior
to enlarging the stack.

Fixes #2187

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
Thanks! :+1:

**Review by wgtdkp — APPROVED** [review]
👍

**Review by andy31415 — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

File: examples/platform/nrf528xx/app/project_include/FreeRTOSConfig.h [hunk 1]
```diff
@@ -97,7 +95,9 @@
 #define configUSE_TIMERS 1
 #define configTIMER_TASK_PRIORITY (2)
 #define configTIMER_QUEUE_LENGTH 32
-#define configTIMER_TASK_STACK_DEPTH (256)
+
+// TODO: Fix high stack usage in GenericThreadStackManagerImpl_FreeRTOS
```
Comment (woody-apple) [inline examples/platform/nrf528xx/app/project_include/FreeRTOSConfig.h:99]: Can we file an issue here?

**Review by woody-apple — APPROVED** [review]

File: examples/platform/nrf528xx/app/project_include/FreeRTOSConfig.h (hunk 1 above)
Comment (mspang) [reply examples/platform/nrf528xx/app/project_include/FreeRTOSConfig.h:99]: There's already an issue and even another PR for reducing it 
https://github.com/project-chip/connectedhomeip/issues/2188

**Comment by gjc13** [comment]
@mspang Is it still required after https://github.com/project-chip/connectedhomeip/pull/2199?

**Comment by mspang** [comment]
> @mspang Is it still required after #2199?

This patch contains all the diagnostic enhancements; we should land at least those. I'm a bit busy to retest it, I think we should land this as-is and then make a followup to set it back to 1k. That way if there's still an overrun at 1k we can just revert the followup.

**Comment by mspang** [comment]
@saurabhst @BroderickCarlin ?

**Comment by mspang** [comment]
This appears to still be broken.

**Comment by mspang** [comment]
Actually, I'm seeing a different issue now

**Comment by mspang** [comment]
@jelderton ?

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**